cmd/main.go

```diff
@@ -199,14 +231,25 @@ func main() {
 		options = append(options, thriftcheck.WithLogger(logger))
 	}
 
-	if len(flag.Args()) == 0 {
+	if len(flag.Args()) == 0 && *stdinFilename == "stdin" {
 		flag.Usage()
 		os.Exit(0)
 	}
 
+	var inputs []string
+	if *stdinFilename != "stdin" {
+		inputs = append(inputs, *stdinFilename)
+	}
```

Comment:
Can you explain this special handling for `stdinFilename`? It's only used as a human-friendly filename when the file content is provided on stdin (via `-`).

Reply:
This is mainly to be compatible with your original parameters, to prevent users from having already written commands in CI or other places. Or this change would become a breaking change. Of course, we could also make a breaking change directly, it depends on whether we want to maintain backward compatibility.

Reply:
I'm still unclear how the `stdinFilename` relates to path expansion. Here's the intended behavior:

* If `-` is passed as the only "filename" argument, then we read from `os.Stdin`.
* We use `stdinFilename` to control the name of that virtual buffer. It defaults to "stdin" but can be customized using the `--stdin-filename` flag.

The condition above is checking to see if that name was customized and then treating the customized "stdin" name as a filename, and that doesn't sound right to me.

Perhaps you meant to check for `-`? But I think we only want to allow and use `-` if it's the _only_ filename argument.

Reply:
It seems like I misunderstood the expected behavior before. So, is it okay if we handle it like this?

- If the user enters `-`, then we read the data to check from `os.Stdin`.
- If the user enters the `--stdin-filename` parameter, then we use its value as a single file path (even if they enter other non-flags), so this ensures backward compatibility of the program.
- Except for the above two cases, we use the new function to recursively obtain the pathList.

Reply:
That's correct. And specifically for your second point, that's already covered explicitly here:

```go
func lint(l *thriftcheck.Linter, filenames []string) (thriftcheck.Messages, error) {
	if len(filenames) == 1 && filenames[0] == "-" {
		return l.Lint(os.Stdin, *stdinFilename)
	}
	return l.LintFiles(filenames)
}
```

Looking at it again, I think this might be the best place to expand the list of input paths. Something like this:

```go
func lint(l *thriftcheck.Linter, paths []string) (thriftcheck.Messages, error) {
	if len(paths) == 1 && paths[0] == "-" {
		return l.Lint(os.Stdin, *stdinFilename)
	}
        filenames, err := expandPaths(paths)
        if err != nil {
            return nil, err
        }
	return l.LintFiles(filenames)
}
```